### PR TITLE
Feature/four 8901: Log Any changes UPDATE within the admin/customize-ui -> Alternative Text and Login Page Footer

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps
Register some events related to user activity events according to the [PRD](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/3084320783/User+Activity+Logging).
Log Any changes UPDATE within the admin/customize-ui -> Alternative Text and Login Page Footer

## Solution
- Improve the tickets definition

## How to Test
Needs to make the following actions:
- Changes in Custom Ui

## Related Tickets & Packages
[- Link to any related FOUR tickets, PRDs, or packages](https://processmaker.atlassian.net/browse/FOUR-8901)

## Code Review Checklist
- [x] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [x] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [x] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [x] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [x] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [x] This ticket conforms to the PRD associated with this part of ProcessMaker.

## Enviroment
ci:deploy

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/CssOverrideController.php [hunk 1]
```diff
@@ -107,8 +107,11 @@ public function store(Request $request)
             $changes = (array)json_decode($setting->getChanges()['config']);
         }
         
+        $changes = array_merge(isset($footer['changes']) ? $footer['changes'] : [], isset($altText['changes']) ? $altText['changes'] : [], $changes);
+        
         if (!empty($changes)) {
-            event(new CustomizeUiUpdated($original, array_merge($footer, $altText, $changes), $setting->getChanges()['updated_at']));
+            $original['config'] = array_merge(isset($footer['original']) ? $footer['original'] : [], isset($altText['original']) ? $altText['original'] : [], isset($original['config']) ? $original['config'] : []);
```
Comment (fagubla) [inline ProcessMaker/Http/Controllers/Api/CssOverrideController.php:113]: This line is too long, could you split in multiple lines?

File: ProcessMaker/Http/Controllers/Api/CssOverrideController.php [hunk 2]
```diff
@@ -107,8 +107,11 @@ public function store(Request $request)
             $changes = (array)json_decode($setting->getChanges()['config']);
         }
         
+        $changes = array_merge(isset($footer['changes']) ? $footer['changes'] : [], isset($altText['changes']) ? $altText['changes'] : [], $changes);
```
Comment (fagubla) [inline ProcessMaker/Http/Controllers/Api/CssOverrideController.php:110]: This line is too long, could you split in multiple lines?

**Review by fagubla — CHANGES_REQUESTED** [review]

**Review by fagubla — APPROVED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/CssOverrideController.php [hunk 3]
```diff
@@ -107,8 +107,19 @@ public function store(Request $request)
...
+        $changes = array_merge(
+            isset($footer['changes']) ? $footer['changes'] : [],
+            isset($altText['changes']) ? $altText['changes'] : [],
+            $changes
+        );
+        
         if (!empty($changes)) {
-            event(new CustomizeUiUpdated($original, array_merge($footer, $altText, $changes), $setting->getChanges()['updated_at']));
+            $original['config'] = array_merge(
+                isset($footer['original']) ? $footer['original'] : [],
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/CssOverrideController.php:118]: Please use the shorthand comparison:
`$footer['original'] ?? [],`

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/CssOverrideController.php [hunk 4]
```diff
@@ -130,7 +143,10 @@ private function setLoginFooter(Request $request)
         $response = [];
 
         if ((!$setting->wasRecentlyCreated && $setting->wasChanged()) || $setting->wasRecentlyCreated) {
-            $response = ['html' => $setting->getAttributes()['config']['html']];
+            $response = [
+                'changes' => ['loginFooter' => $setting->getAttributes()['config']['html']],
+                'original' => ['loginFooter' => isset($original) ? $original : '']
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/CssOverrideController.php:148]: Please use this:
`'original' => ['loginFooter' => $original ?? '']`

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Review by pmPaulis — APPROVED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

File: ProcessMaker/Http/Controllers/Api/CssOverrideController.php [hunk 5]
```diff
@@ -121,6 +132,9 @@ private function setLoginFooter(Request $request)
             $footerContent = '';
         }
 
+        $original = Setting::where('key', 'login-footer')->first();
+        $original = isset($original) ? $original->getAttribute('config')['html'] : '';
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/CssOverrideController.php:136]: Please validate if ['html'] exist

File: ProcessMaker/Http/Controllers/Api/CssOverrideController.php [hunk 6]
```diff
@@ -130,7 +144,10 @@ private function setLoginFooter(Request $request)
         $response = [];
 
         if ((!$setting->wasRecentlyCreated && $setting->wasChanged()) || $setting->wasRecentlyCreated) {
-            $response = ['html' => $setting->getAttributes()['config']['html']];
+            $response = [
+                'changes' => ['loginFooter' => $setting->getAttributes()['config']['html']],
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/CssOverrideController.php:148]: Please validate if ['html'] exist the use
$setting->getAttribute('config') insteand of $setting->getAttributes()['config']

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://ci-a6841ab6d4.eng.processmaker.net

**Pull request closed without merging**